docs/bundler/migration.md

```diff
@@ -0,0 +1,657 @@
+
+In the end,
+
+## CLI
+
+Bun and esbuild both provide a command-line interface.
+
+```bash
+$ esbuild <entrypoint> --bundle --outdir=out --bundle
+$ bun build <entrypoint> --outdir=out
+```
+
+There are a few behavioral differences to note.
+
+- Bun _always bundles by default_. This is why the `--bundle` flag isn't necessary in the Bun example.
+- Bun's bundler does not include a built-in development server or file watcher. It's just a bundler. The bundler can be used in conjunction with `Bun.serve`, `--watch`, and other built-in Bun functionality to achieve the same effect. As such, all options relating to HTTP/file watching are not applicable.
+
+{% table %}
+
+- `esbuild`
+- `bun build`
+
+---
+
+- `--bundle`
+- n/a
+- Not necessary, `bun build` always bundles.
+
+---
+
+- `--define:K=V`
+- `--define:K=V`
+- No differences
+
+---
+
+- `--external`
+- `--external`
+- No differences
+
+---
+
+- `--format`
+- `--module`
+
+---
+
+- `--loader`
+- `--loader`
+- Slight syntactic difference for compatibility with `bun run`.
+
+  ```bash
+  $ esbuild app.ts --bundle --loader:.svg=text
+  $ bun build app.ts --loader .svg:text
+  ```
+
+---
+
+- `--minify`
+- `--minify`
+- No differences
+
+---
+
+- `--outdir`
+- `--outdir`
+- No differences
+
+---
+
+- `--outfile`
+- `--outfile`
+
+---
+
+- `--packages`
+- n/a
+- Not supported
+
+---
+
+- `--platform`
+- `--target`
+- Renamed to `--target` for consistency with tsconfig
+
+---
+
+- `--serve`
+- n/a
+- Not applicable.
+
+---
+
+- `--sourcemap`
+- `--sourcemap`
+- No differences
+
+---
+
+- `--splitting`
+- `--splitting`
+- No differences
+
+---
+
+- `--target`
+- n/a
+- No supported. Bun's bundler performs no syntactic downleveling at this time.
+
+---
+
+- `--watch`
+- n/a
+- Not applicable.
+
+---
+
+- `--allow-overwrite`
+- n/a
+- Overwriting is always allowed
+
+---
+
+- `--analyze`
+- n/a
+- Not supported. Use `--manifest` to generate a manifest file.
+
+---
+
+- `--asset-names`
+- `--asset-naming`
+- Renamed for consistency with `naming` in JS API
+
+---
+
+- `--banner`
+- n/a
+- Not supported
+
+---
+
+- `--certfile`
+- n/a
+- Not applicable, Bun's bundler does
+
+---
+
+- `--charset=utf8`
+- n/a
+- Not supported
+
+---
+
+- `--chunk-names`
+- `--chunk-naming`
+- Renamed for consistency with `naming` in JS API
+
+---
+
+- `--color`
+- n/a
+- Always enabled
+
+---
+
+- `--drop`
+- n/a
+- Not supported
+
+---
+
+- `--entry-names`
+- `--entry-naming`
+- Renamed for consistency with `naming` in JS API
+
+---
+
+- `--footer`
+- n/a
+- Not supported
+
+---
+
+- `--global-name`
+- n/a
+- Not applicable, Bun does not support `iife` output at this time.
+
+---
+
+- `--ignore-annotations`
+- n/a
+- Not supported
+
+---
+
+- `--inject`
+- n/a
+- Not supported
+
+---
+
+- `--jsx`
+- n/a
+- Not supported
+
+---
+
+- `--jsx-dev`
+- n/a
+- Bun uses `jsxDEV` by default. Use `--jsx-production` to use `jsx`.
```

Comment:
--jsx-production doesnt work. this might be a bug though

`NODE_ENV=production` to use the production `jsx` runtime (as well as set react itself into production mode)